src/components/product-card/HorizontalCard/ProductCardHorizontal.jsx

```diff
@@ -44,49 +44,153 @@ const ProductCardHorizontal = ({ item }) => {
     }
   }
 
+  async function removeFromWishlist(product) {
+    try {
+      const responseWishlist = await axios.get(`/api/user/wishlist`, {
+        headers: {
+          authorization: loginToken,
+        },
+      });
+      if (
+        responseWishlist.data.wishlist.find((item) => item._id === product._id)
+      ) {
+        const response = await axios.delete(
+          `/api/user/wishlist/${product.id}`,
+          {
+            headers: {
+              authorization: loginToken,
+            },
+          }
+        );
+        if (response.status === 200 || response.status === 201) {
+          const serverResponse = await axios.get(`/api/user/wishlist`, {
+            headers: {
+              authorization: loginToken,
+            },
+          });
+          dispatch({
+            type: "LOAD_WISHLIST",
+            payload: serverResponse.data.wishlist,
+          });
+        }
+      }
+    } catch (e) {
+      console.log(e);
+    }
+  }
+
+  async function addProductToCart(product) {
+    try {
+      const response = await axios.post(
+        "/api/user/cart",
+        {
+          product,
+        },
+        {
+          headers: {
+            authorization: loginToken,
+          },
+        }
+      );
+      if (response.status === 200 || response.status === 201) {
+        dispatch({
+          type: "LOAD_CART",
+          payload: response.data.cart,
+        });
+      }
+    } catch (e) {
+      console.log(e);
+    }
+  }
+
+  async function removeFromCart(product) {
+    try {
+      const responseCart = await axios.get(`/api/user/cart`, {
+        headers: {
+          authorization: loginToken,
+        },
+      });
+      if (responseCart.data.cart.find((item) => item._id === product._id)) {
```

Comment:
Suggestion: you can eleminate one api call by using cart data in context.

Reply:
I will fix these changes in next feature commit